### PR TITLE
Mission outcome processing

### LINKED ISSUE #53: Make missions use outcome/modifier data models
At the moment, the data models are not used. Rewards are not granted to the player after completing missions. Ship damage is not dealt. etc. 

We already have scriptable objects set up to handle these outcomes, but the code needs to be updated to use them.

### PR DESCRIPTION
**Pull request by maxwellpark**

### CODE REVIEW

File: Assets/Scripts/DataModels/Missions/MissionOutcomes/MoneyOutcome.cs [hunk 1]
```diff
@@ -6,7 +6,7 @@ public class MoneyOutcome : MissionOutcome
     public float moneyMin;
     public float moneyMax;
 
-    public override void Process()
+    public override void Process(Mission mission)
```
Comment (maxwellpark) [inline Assets/Scripts/DataModels/Missions/MissionOutcomes/MoneyOutcome.cs:9]: Yeah I wanted to overload this and have a parameterless one. But couldn't find a clean way of determining which method to call in Mission.ProcessOutcome

File: Assets/Scripts/DataModels/Pilots/PilotAccessors.cs [hunk 2]
```diff
@@ -38,7 +39,10 @@ public bool IsAssignedToShip
         get => saveData.isAssignedToShip; set { saveData.isAssignedToShip = value; }
     }
 
-    public Ship Ship { get => ship; }
+    public Ship Ship 
+    {
+        get => ShipsManager.Instance.Ships.FirstOrDefault(s => s.Pilot == this);
```
Comment (maxwellpark) [inline Assets/Scripts/DataModels/Pilots/PilotAccessors.cs:44]: We don't want the Pilot model having a Ship on it while the Ship has a Pilot on it (circular). So, the Ship field has been deleted on the Pilot and replaced with this. Probably a better way.

File: Assets/Scripts/Extensions/ShipExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,11 @@
+﻿public static class ShipExtensions
+{
+    public static void DeductFuel(this Ship self)
```
Comment (maxwellpark) [inline Assets/Scripts/Extensions/ShipExtensions.cs:3]: Did this just for fun

File: Assets/Scripts/DataModels/Pilots/PilotAccessors.cs (hunk 2 above)
Comment (ambid17) [reply Assets/Scripts/DataModels/Pilots/PilotAccessors.cs:44]: i couldn't figure out a better way than the circular reference :(

File: Assets/Scripts/DataModels/Missions/MissionOutcomes/MoneyOutcome.cs (hunk 1 above)
Comment (ambid17) [reply Assets/Scripts/DataModels/Missions/MissionOutcomes/MoneyOutcome.cs:9]: yeah this is fine for now

**Review by ambid17 — APPROVED** [review]

File: Assets/Scripts/DataModels/Missions/Mission.cs [hunk 4]
```diff
@@ -53,7 +53,10 @@ public void ProcessOutcomes()
     {
         foreach (MissionOutcome outcome in outcomes)
         {
-            outcome.Process(this);
+            if (outcome != null)
+            {
+                outcome.Process(this);
```
Comment (maxwellpark) [inline Assets/Scripts/DataModels/Missions/Mission.cs:58]: If we overload the Process method, then when we loop through the outcomes, how do we know whether to call the one with the Mission param vs. the other one? Some outcomes need to get the Mission and others don't.

Checking types and casting them or something seems ridiculous.

Could we just have the void method? Well maybe, but we need a way of referencing the Mission scriptable object from the individual outcomes. Seems kinda hard as theres no inheritance there, and looking them up in the database or something seems ridiculous too.

**Pull request merged**